### ISSUE TITLE
The interpreter is #265-ing

### ISSUE THREAD

**Issue by KristofferC** [JuliaDebug/JuliaInterpreter.jl#262]
```jl
julia> f(x) = x
f (generic function with 1 method)

julia> g(x) = f(x)
g (generic function with 1 method)

julia> @interpret g(5)
5

julia> g(5)
5

julia> f(x) = x*x
f (generic function with 1 method)

julia> @interpret g(5)
5

julia> g(5)
25
```

**Comment by timholy** [comment]
With late-binding breakpoints we could go back to having `@interpret` clear the `framedict`, `genframedict` caches prior to each call. Alternatively, Revise can fix this (but that doesn't help if Revise isn't running).

**Comment by KristofferC** [comment]
Yeah, similar as to how we empty `junk!` on "entrance from toplevel" we could also empty the caches. I wonder what the performance impact is on second call vs having all those FrameCodes cached. 

And yeah, there are ways to create a frame without cause the caches to clear but we could perhaps just document that the caches should be cleared if one want updated FrameCode's or something.

**Comment by KristofferC** [comment]
Also, Revise cannot fix this with manual redefinitions like in the first post, or?

**Comment by timholy** [comment]
Yes, it could, since it currently scans the REPL inputs for method definitions and adds them to a fake `@REPL` "package". I could add logic that would look for entries in `framedict` with the same signature. Just not sure whether we want to tie that functionality to Revise.

**Comment by KristofferC** [comment]
Okay, but say like Juno Ctrl-enter evals. Anyway, bumping the "world age" on entrance from top level seems like the correct thing to do. I'll see how it influences timings of e.g. Plots.

**Comment by KristofferC** [comment]
```jl
julia> @time @interpret plot(rand(5));
  1.795993 seconds (13.40 M allocations: 444.933 MiB, 5.35% gc time)

julia> @time @interpret plot(rand(5));
  1.802492 seconds (13.40 M allocations: 444.933 MiB, 5.58% gc time)

julia> JuliaInterpreter.empty!(JuliaInterpreter.framedict)
Dict{Method,JuliaInterpreter.FrameCode} with 0 entries

julia> JuliaInterpreter.empty!(JuliaInterpreter.genframedict)
Dict{Tuple{Method,Type},JuliaInterpreter.FrameCode} with 0 entries

julia> @time @interpret plot(rand(5));
  2.216476 seconds (14.19 M allocations: 487.105 MiB, 5.75% gc time)

julia> JuliaInterpreter.empty!(JuliaInterpreter.genframedict)
Dict{Tuple{Method,Type},JuliaInterpreter.FrameCode} with 0 entries

julia> JuliaInterpreter.empty!(JuliaInterpreter.framedict)
Dict{Method,JuliaInterpreter.FrameCode} with 0 entries

julia> @time @interpret plot(rand(5));
  2.253128 seconds (14.19 M allocations: 487.099 MiB, 6.09% gc time)
```

So 0.5 seconds overhead. Not too bad...?

**Comment by timholy** [comment]
I agree that the best solution is to check world age. The second is to clear the cache, once #259 is merged and we're happy with it and not worried about any weird implications.

**Comment by KristofferC** [comment]
So the path forward is to focus on #259, get that merged and then clear caches?

**Comment by timholy** [comment]
I think that's quite reasonable. Eventually we will want to do the whole world-age thing, but that's a bigger project.

**Comment by KristofferC** [comment]
I am quite sad this is not issue number 265.

**Comment by timholy** [comment]
And so close, too. In Revise there was an almost-related concern (double-evaluation rather than re-evaluation), though it was a PR rather than an issue: https://github.com/timholy/Revise.jl/pull/265

I find it amazing that we're already had this many PRs/issues; Revise is 1.5yrs old and is only barely higher. We have been busy here!

**Comment by KristofferC** [comment]
I thought about waiting with https://github.com/timholy/CodeTracking.jl/issues/38 until we reached 265 but that would probably have taken too long.

**Comment by KristofferC** [comment]
Closed by #259
